src/Gir/Marshal/Array.cs

```diff
@@ -5,7 +5,12 @@ public partial class Array
 	{
 		public ISymbol GetSymbol (GenerationOptions opts)
 		{
-			return opts.SymbolTable [CType];
+			if (Name != null) {
```

Comment:
Sometimes there is for some reason no Name on arrays. Bug in gi maybe?

Reply:
Example?

Reply:
Are you sure you haven't modified the gir locally? It does have it for me.

Reply:
```
    <function name="byte_array_new_take"
              c:identifier="g_byte_array_new_take"
              moved-to="ByteArray.new_take"
              version="2.32">
      <doc xml:space="preserve">Create byte array containing the data. The data will be owned by the array
and will be freed with g_free(), i.e. it could be allocated using g_strdup().</doc>
      <return-value transfer-ownership="full">
        <doc xml:space="preserve">a new #GByteArray</doc>
        <array name="GLib.ByteArray" c:type="GByteArray*">
          <type name="guint8" c:type="guint8"/>
        </array>
      </return-value>
      <parameters>
        <parameter name="data" transfer-ownership="full">
          <doc xml:space="preserve">byte data for the array</doc>
          <array length="1" zero-terminated="0" c:type="guint8*">
            <type name="guint8" c:type="guint8"/>
          </array>
        </parameter>
        <parameter name="len" transfer-ownership="none">
          <doc xml:space="preserve">length of @data</doc>
          <type name="gsize" c:type="gsize"/>
        </parameter>
      </parameters>
    </function>
```
It's not there in the data parameter

Reply:
Well, for that case, I looked at the code, and in that case, it's just a uint8 pointer:
https://github.com/GNOME/glib/blob/90d3337e1d9d8218f1f8509417b6ae65993aa029/glib/garray.c#L1661

I guess when name isn't there, we need to pass in an array of the element type, it's not a specific gtype

Reply:
So, when we have a array with no name, it's just an `Array` ISymbol that has `T[]` as the managed type